bindings/src/index.ts

```diff
@@ -654,10 +445,27 @@ class NEARBindingsBuilder extends BaseVisitor {
 
 }
 
+function createEncodeStatements(_class: ClassDeclaration): string[] {
+  return _class.members.filter(isField).map(
+    (field: FieldDeclaration): string  => {
+      let T = toString(field.type!);
+      let name = toString(field.name);
+      return `encode<${T}>(encoder, this.${name}, "${name}");`;
+    }
+  );
+}
+// TODO: Make work for non-simple generics e.g. field: Array<T>
```

Comment:
Is this `TODO` still relevant?

Reply:
Yes. Consider

```ts
class Faa<T>{
  t:T
}

class Foo<T> {
  faa: Faa<T>;
}
```

Our current way of determining if `arr` has a generic type would not find this.  It is a simple fix, which is why I left it as a TODO.